### ISSUE TITLE
A default absence of mapgen spflags cannot be edited in settings

### ISSUE THREAD

**Issue by paramat** [minetest/minetest#3356]
@PilzAdam 
Both fractal and flat mapgens have options that are unset by default, so they have a blank spflags field in settings. If in settingtypes.txt the 'default' is left blank, with 2 spaces between 'type' and 'possible values' as advised, the flags cannot be edited in settings.
So currently i am setting the flat mapgen spflags as 'nolakes,nohills', which is possibly not ideal as 'no' is usually used to unset a flag that is set by default. This needs sorting out before we can unhide flat or fractal mapgens.

**Comment by PilzAdam** [comment]
Sorry, I don't understand. What exactly is your issue here?

**Comment by paramat** [comment]
If the spflags field is blank by default settingtypes looks like this:

mgfractal_spflags (Mapgen fractal flags) flags  julia,nojulia

On your earlier advice there are 2 spaces between 'type' and 'possible flags'.
However in-game when you try to edit the spflags field to 'julia' to enable that option, you can't because the spflags field doesn't show up at all in the settings menu.

So currently to make this work the line in settingtypes is:

mgfractal_spflags (Mapgen fractal flags) flags nojulia julia,nojulia

Which may not be acceptable because 'nojulia' is usually only used to disable an option that is enabled by default.

**Comment by paramat** [comment]
By the way, setting mapgen flags in .conf is broken, perhaps this was intentional? setting them in the settings menu works correctly though.

**Comment by PilzAdam** [comment]
Ah, so the actual bug is that settings with an empty default string are not shown in the GUI.

**Comment by paramat** [comment]
Correct.

**Comment by PilzAdam** [comment]
Are there any error messages printed about it?

**Comment by paramat** [comment]
No.

**Comment by paramat** [comment]
@PilzAdam please can this be fixed before next release? (3 weeks) it's the only issue delaying mgflat.

**Comment by PilzAdam** [comment]
@paramat yes; I will probably find some time for this in the next 3 weeks.

**Comment by PilzAdam** [comment]
>> Are there any error messages printed about it?
>
> No.

If I add this to `settingtypes.txt`
> `mgfractal_spflags (Mapgen fractal flags) flags  julia,nojulia`

 I get the following error:
```
ERROR[Main]: Invalid flags setting in /home/adam/Games/Minetest/minetest/bin/../builtin/settingtypes.txt "mgfractal_spflags (Mapgen fractal flags) flags  julia,nojulia"
```

There is no way a line in `settingtypes.txt` is not added to the menu without error messages.

Anyway, fix is in #3402.
